pyproject.toml

```diff
@@ -43,12 +43,12 @@ cmake.source-dir = "src"
 cmake.targets = ["skbuild-export"]
 # TODO: This should be removed, but currently can't bacause: 'src/src/CMakeLists.txt'
 sdist.include = [
-    ".git"
+    ".git",
```

Comment:
Hmm, I would rather not include this in the sdist. Please try this instead: https://github.com/TileDB-Inc/TileDB-Py/blob/ff2149738a5daeee2c11d7660b3c7631fe1a293e/setup.py#L809

Reply:
I would also like to remove it from there, but the reason why I left .git in this include is that the CMake that we use actually use it. Not setuptools_scm. And that logic in cmake breaks during sdist build, since it looks for .git folder. However, I just noticed that that git logic got moved to `logging.cmake` and it might not even be used anymore. I'll try `sdist` and if it works I'll remove that .git folder from package.

Reply:
```
      FAILED: libtiledbvectorsearch-prefix/src/libtiledbvectorsearch-stamp/libtiledbvectorsearch-configure /tmp/tmpji08b7ja/build/libtiledbvectorsearch-prefix/src/libtiledbvectorsearch-stamp/libtiledbvectorsearch-configure
      cd /tmp/tmpji08b7ja/build/libtiledbvectorsearch && /usr/bin/cmake -DSUPERBUILD=OFF -DCMAKE_INSTALL_PREFIX=/usr/local "-DCMAKE_PREFIX_PATH=/home/dudoslav/.miniforge3/envs/zoo/lib/python3.11/site-packages;/tmp/pip-build-env-0umpripn/overlay/lib/python3.11/site-packages;" -DCMAKE_MODULE_PATH=/tmp/pip-req-build-9ud69_n7/src/cmake/Modules/ /tmp/pip-req-build-9ud69_n7/src/cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_SYSTEM_PROCESSOR=x86_64 -DCMAKE_OSX_ARCHITECTURES= -DCMAKE_C_COMPILER=/usr/bin/cc -DCMAKE_C_FLAGS= -DCMAKE_CXX_COMPILER=/usr/bin/c++ -DCMAKE_CXX_FLAGS= -DCMAKE_CXX_STANDARD=20 -DEP_BASE=/tmp/tmpji08b7ja/build/externals -DFORCE_EXTERNAL_TILEDB=OFF -DTILEDB_S3=ON -DTileDB_DIR= -DOVERRIDE_INSTALL_PREFIX=OFF -DBLA_VENDOR= -DUSE_MKL_CBLAS=OFF -DTILEDB_VS_PYTHON=ON -DBUILD_TESTS=ON -DPython_FIND_REGISTRY=NEVER -DPYTHON_VERSION_STRING= -DPYTHON_LIBRARY= -DPYTHON_INCLUDE_DIR=/home/dudoslav/.miniforge3/envs/zoo/include/python3.11 -DPython_EXECUTABLE=/home/dudoslav/.miniforge3/envs/zoo/bin/python3.11 -DPython_ROOT_DIR=/home/dudoslav/.miniforge3/envs/zoo -DPython_INCLUDE_DIR=/home/dudoslav/.miniforge3/envs/zoo/include/python3.11 -DPython3_FIND_REGISTRY=NEVER -DPython3_INCLUDE_DIR=/home/dudoslav/.miniforge3/envs/zoo/include/python3.11 -Dpybind11_DIR= -DEP_TILEDB_BUILT=TRUE -GNinja -S /tmp/pip-req-build-9ud69_n7/src -B /tmp/tmpji08b7ja/build/libtiledbvectorsearch && /usr/bin/cmake -E touch /tmp/tmpji08b7ja/build/libtiledbvectorsearch-prefix/src/libtiledbvectorsearch-stamp/libtiledbvectorsearch-configure
      CMake Warning:
        Ignoring extra path from command line:
      
         "/tmp/pip-req-build-9ud69_n7/src/cmake"
      
      
      CMake Warning:
        Ignoring extra path from command line:
      
         "/tmp/pip-req-build-9ud69_n7/src/cmake"
      
      
      -- The C compiler identification is GNU 13.2.1
      -- The CXX compiler identification is GNU 13.2.1
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /usr/bin/cc - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /usr/bin/c++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Starting TileDB-Vectorsearch regular build.
      Architecture is x86_64
      -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
      -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
      -- Found Threads: TRUE
      -- TileDB_DIR is set to  -- find_package will search there first.
      -- Found TileDB: /tmp/tmpji08b7ja/build/externals/install/lib/libtiledb.so.2.20
      CMake Deprecation Warning at /tmp/tmpji08b7ja/build/libtiledbvectorsearch/_deps/docopt-src/CMakeLists.txt:1 (cmake_minimum_required):
        Compatibility with CMake < 3.5 will be removed from a future version of
        CMake.
      
        Update the VERSION argument <min> value or use a ...<max> suffix to tell
        CMake that the project does not need compatibility with older versions.
      
      
      -- Fetching mdspan...
      -- Detected support for C++23 standard
      -- Performing Test COMPILER_SUPPORTS_FCONCEPTS
      -- Performing Test COMPILER_SUPPORTS_FCONCEPTS - Success
      -- Using "-fconcepts" to enable concepts support
      Installed json not found -- fetching
      CMake Deprecation Warning at /tmp/tmpji08b7ja/build/libtiledbvectorsearch/_deps/json-src/CMakeLists.txt:1 (cmake_minimum_required):
        Compatibility with CMake < 3.5 will be removed from a future version of
        CMake.
      
        Update the VERSION argument <min> value or use a ...<max> suffix to tell
        CMake that the project does not need compatibility with older versions.
      
      
      -- Using the single-header code from /tmp/tmpji08b7ja/build/libtiledbvectorsearch/_deps/json-src/single_include/
      fatal: not a git repository (or any parent up to mount point /)
      Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
      fatal: not a git repository (or any parent up to mount point /)
      Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
      CMake Error at cmake/logging.cmake:28 (get_filename_component):
        get_filename_component called with incorrect number of arguments
      Call Stack (most recent call first):
        CMakeLists.txt:175 (include)
      
      
      -- Configuring incomplete, errors occurred!
      ninja: build stopped: subcommand failed.
      
      *** CMake build failed
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for tiledb-vector-search
Failed to build tiledb-vector-search
ERROR: Could not build wheels for tiledb-vector-search, which is required to install pyproject.toml-based projects
```

Reply:
This is what happens ^^ when I remove `.git` from that repo. It will break `sdist`

Reply:
We know we can generate a version file from setuptools, so I think we should do that and then parse it in cmake to get the version (if this file exists).

Some related discussion here, but the suggestion is a separate repo AFAICT: https://github.com/scikit-build/scikit-build-core/issues/631

Reply:
This is a great observation, and yes, we could use `SKBUILD_PROJECT_VERSION_FULL`. But unfortunately, right now we log these values:
```
constexpr const char GIT_REPO_NAME[] = "@GIT_REPO_NAME@";
constexpr const char GIT_REPO_URL[] = "@GIT_REPO_URL@";
constexpr const char GIT_BRANCH[] = "@GIT_BRANCH@";
constexpr const char GIT_COMMIT_HASH[] = "@GIT_COMMIT_HASH@";
constexpr const char GIT_COMMIT_DATE[] = "@GIT_COMMIT_DATE@";
constexpr const char GIT_COMMIT_TIME[] = "@GIT_COMMIT_TIME@";
```

That are not present in that previously mentioned variable.

Reply:
I would suggest merging this PR in order to fix the versioning ASAP and create a new story just for removing `.git` from sdist.